anvio/variability.py

```diff
@@ -355,7 +351,7 @@ def process(self, *args, **kwargs):
         self.rename_key('competing_items', 'competing_codons')
         self.rename_key('pos', 'codon_order_in_gene')
 
-
+# FIXME: it seems the following classes are redundant and should be removed? Ping @evan
```

Comment:
I would contest the removal of these classes. They are mostly redundant at the moment (who knows what specific operations we may want to perform on specific classes of allele counts in the future), but even in the current state there are trivial lines of code (for column renaming) that if not put here, are pushed to a module that should have no business changing column names. So my main points are (1) this structure allows for item-specific extensions and modifications, (2) maintains separation of code responsibility. If you still think we should remove these classes, I can move these column renaming operations elsewhere.

Reply:
totally fine. I didn't look into it carefully, just caught my eye and wanted to mention it somewhere.